### PR TITLE
Search clear

### PR DESCRIPTION
**Pull request by MieczyslawFrontDev**
Hi there !

I created new functionality that cleaning search content. 
I created new components and I used flex technology to position elements inside SearchBox component.
I put all search related components into the Search component to organize the structure of the application.
Also I moved SVG icons from Icons folder to Images.

### CODE REVIEW

File: src/common/Search/index.js [hunk 1]
```diff
@@ -22,20 +46,31 @@ export const Search = () => {
...
-    />
+          disabled={
+            location.pathname.indexOf("/movies/") === 0 ||
+            location.pathname.indexOf("/people/") === 0
+          }
+        />
+      </StyledSearchContainer>
+      <ClearContentButton onClick={clearInputContent}>
+        <ClearContentIcon />
+      </ClearContentButton>
```
Comment (paulgrym) [inline src/common/Search/index.js:72]: I would add a condition to render the clear button only when query exists :)

File: src/common/Search/styled.js [hunk 2]
```diff
@@ -20,12 +58,30 @@ export const StyledSearch = styled.input`
...
     font-size: 13px;
-    padding: 14px 0 14px 40px;
   }
 
   @media (max-width: ${({ theme }) => theme.breakpoints.large}) {
     width: 100%;
   }
 `;
+
+export const ClearContentIcon = styled(ClearSVG)`
```
Comment (paulgrym) [inline src/common/Search/styled.js:69]: Maybe adding a fill with a color set to "waterloo" would make "X" look nicer 😃

File: src/common/queryParameterHooks.js [hunk 3]
```diff
@@ -14,14 +14,13 @@ export const useReplaceQueryParameter = () => {
...
-        searchParams.delete(key);
+      if (value === "") {
+        searchParams.delete(key); 
       } else {
         searchParams.set(key, value);
       }
     });
 
-    history.push(`${location.pathname}?${searchParams.toString()}`);
+    history.replace(`${location.pathname}?${searchParams.toString()}`);
```
Comment (paulgrym) [inline src/common/queryParameterHooks.js:24]: If you add a condition to render clear button only when the query exists you won't have to change history.push to history.replace to eliminate warning in the console. 😁

**Review by paulgrym — APPROVED** [review]
Good job! 🎉 As always I have some ideas on what to improve 😀

**Comment by MieczyslawFrontDev** [comment]
I improved the functionality of the application ;)

**Pull request merged**